datafusion/src/logical_plan/builder.rs

```diff
@@ -289,6 +288,37 @@ impl LogicalPlanBuilder {
         }))
     }
 
+    /// Apply a window
+    pub fn window(
+        &self,
+        window_expr: impl IntoIterator<Item = Expr>,
+        // partition_by_expr: impl IntoIterator<Item = Expr>,
+        // order_by_expr: impl IntoIterator<Item = Expr>,
+        // window_frame: Option<WindowFrame>,
+    ) -> Result<Self> {
+        let window_expr = window_expr.into_iter().collect::<Vec<Expr>>();
+        // let partition_by_expr = partition_by_expr.into_iter().collect::<Vec<Expr>>();
+        // let order_by_expr = order_by_expr.into_iter().collect::<Vec<Expr>>();
+        let all_expr = window_expr.iter();
+        validate_unique_names("Windows", all_expr.clone(), self.plan.schema())?;
+
+        let mut window_fields: Vec<DFField> =
+            exprlist_to_fields(all_expr, self.plan.schema())?;
+        window_fields.extend_from_slice(self.plan.schema().fields());
+
+        Ok(Self::from(&LogicalPlan::Window {
+            input: Arc::new(self.plan.clone()),
+            // FIXME implement next
+            // partition_by_expr,
+            // FIXME implement next
+            // order_by_expr,
+            // FIXME implement next
+            // window_frame,
+            window_expr,
```

Comment:
likewise here it may make sense to have a filter clause (FIXME) too

Reply:
I'll leave a comment for now. btw `function.filter` needs to be added in the upstream `sqlparser` crate.

Reply:
added